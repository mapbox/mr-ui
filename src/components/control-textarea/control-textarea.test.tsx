import React from 'react';
import ControlTextarea from './control-textarea';
import { screen, render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('ControlTextarea', () => {
  describe('basic', () => {
    const mockOnChange = jest.fn();
    const props = {
      id: 'testinput',
      label: 'Basic',
      onChange: mockOnChange
    } as const;

    test('renders as expected', () => {
      const { baseElement } = render(<ControlTextarea {...props} />)
      expect(baseElement).toMatchSnapshot();
    });

    test('onChange is called with input id and value', async () => {
      render(<ControlTextarea {...props} />)
      await userEvent.type(screen.getByTestId('testinput-textarea'), 'f');

      await waitFor(() => {
        expect(mockOnChange).toHaveBeenCalledTimes(1);
      });

      await waitFor(() => {
        expect(mockOnChange).toHaveBeenCalledWith('f', 'testinput');
      });
    });
  });

  describe('disabled', () => {
    const mockOnChange = jest.fn();
    const props = {
      id: 'testinput',
      label: 'basic',
      onChange: mockOnChange,
      disabled: true
    } as const;

    test('renders as expected', () => {
      const { baseElement } = render(<ControlTextarea {...props} />)
      expect(baseElement).toMatchSnapshot();
    });

    test('onChange is called with input id and value', async () => {
      render(<ControlTextarea {...props} />)
      await userEvent.type(screen.getByTestId('testinput-textarea'), 'steve');

      await waitFor(() => {
        expect(mockOnChange).toHaveBeenCalledTimes(0);
      });
    });
  });

  describe('all options', () => {
    const props = {
      id: 'testinput',
      label: 'All options',
      validationError: 'oh no!',
      noAuto: true,
      optional: true,
      autoFocus: true,
      aside: <span>Aside text</span>,
      placeholder: 'Some cool placeholder text',
      value: 'Some cool default value',
      themeControlTextarea: 'w-full border--pink round-bold px12 py12',
      themeControlWrapper: 'bg-red px12 py12',
      themeLabel: 'txt-s',
      onChange: jest.fn()
    } as const;

    test('renders as expected', () => {
      const { baseElement } = render(<ControlTextarea {...props} />)
      expect(baseElement).toMatchSnapshot();
    });
  });
});
