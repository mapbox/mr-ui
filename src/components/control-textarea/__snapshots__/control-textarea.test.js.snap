// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlToggleSet allOptions renders 1`] = `
<ControlWrapper
  id="testinput"
  themeControlWrapper="bg-red px12 py12"
  validationError="oh no!"
>
  <ControlLabel
    aside={
      <span>
        Aside text
      </span>
    }
    id="testinput"
    optional={true}
    text="All options"
    themeLabel="txt-s"
  />
  <textarea
    aria-invalid={true}
    aria-required={false}
    autoCapitalize="none"
    autoCorrect="off"
    autoFocus={true}
    className="w-full border--pink round-bold px12 py12"
    data-testid="testinput-textarea"
    id="testinput"
    name="testinput"
    onChange={[Function]}
    placeholder="Some cool placeholder text"
    spellCheck={false}
    value="Some cool default value"
  />
</ControlWrapper>
`;

exports[`ControlToggleSet basic onChange gets called with input id and value 1`] = `
<ControlWrapper
  id="testinput"
>
  <ControlLabel
    id="testinput"
    optional={false}
    text="Basic"
  />
  <textarea
    aria-required={true}
    className="textarea hmin120"
    data-testid="testinput-textarea"
    id="testinput"
    name="testinput"
    onChange={[Function]}
    value=""
  />
</ControlWrapper>
`;

exports[`ControlToggleSet basic renders 1`] = `
<ControlWrapper
  id="testinput"
>
  <ControlLabel
    id="testinput"
    optional={false}
    text="Basic"
  />
  <textarea
    aria-required={true}
    className="textarea hmin120"
    data-testid="testinput-textarea"
    id="testinput"
    name="testinput"
    onChange={[Function]}
    value=""
  />
</ControlWrapper>
`;

exports[`ControlToggleSet disabled renders 1`] = `
<ControlWrapper
  id="testinput"
>
  <ControlLabel
    id="testinput"
    optional={false}
    text="Disabled"
  />
  <textarea
    aria-required={true}
    className="textarea hmin120"
    data-testid="testinput-textarea"
    disabled={true}
    id="testinput"
    name="testinput"
    onChange={[Function]}
    value=""
  />
</ControlWrapper>
`;
